everware/spawner.py

```diff
@@ -220,17 +195,16 @@ def build_image(self):
                 return image_name
 
         tmp_dir = mkdtemp(suffix='-everware')
+        self._git_executor = GitExecutor(self.form_repo_url, tmp_dir)
         self._add_to_log('Cloning repository %s' % self.repo_url)
         self.log.info('Cloning repo %s' % self.repo_url)
-        yield self.git('clone', self.repo_url, tmp_dir)
+        yield self._git_executor.exec()
         # use git repo URL and HEAD commit sha to derive
         # the image name
-        repo = git.Repo(tmp_dir)
-        self.repo_sha = repo.rev_parse("HEAD")
 
         image_name = "everware/{}-{}".format(
-            self.escaped_repo_url,
-            self.repo_sha
+            self._git_executor.escaped_repo_url,
```

Comment:
`self.excaped_repo_url` if we add this as a shortcut like we have for `repo_url` and `commit_sha`